code/datums/components/creamed.dm

```diff
@@ -0,0 +1,82 @@
+GLOBAL_LIST_INIT(creamable, typecacheof(list(
+	/mob/living/carbon/human,
+	/mob/living/carbon/monkey,
+	/mob/living/simple_animal/pet/dog/corgi,
+	/mob/living/silicon/ai)))
+
+/**
+  * # Creamed component
+  *
+  * For when you have pie on your face
+  */
+/datum/component/creamed
+	dupe_mode = COMPONENT_DUPE_UNIQUE_PASSARGS
+
+	var/mutable_appearance/creamface
+
+/datum/component/creamed/Initialize(datum/source, stunning)
+	if(!isliving(parent))
+		return COMPONENT_INCOMPATIBLE
+
+	if(is_type_in_typecache(parent, GLOB.creamable))
+		creamface = mutable_appearance('icons/effects/creampie.dmi')
+
+		if(ishuman(parent))
+			var/mob/living/carbon/human/H = parent
+			if(H.dna.species.limbs_id == "lizard")
+				creamface.icon_state = "creampie_lizard"
+			else
+				creamface.icon_state = "creampie_human"
+			SEND_SIGNAL(H, COMSIG_ADD_MOOD_EVENT, "creampie", /datum/mood_event/creampie)
+		else if(ismonkey(parent))
+			creamface.icon_state = "creampie_monkey"
+		else if(iscorgi(parent))
+			creamface.icon_state = "creampie_corgi"
+		else if(isAI(parent))
+			creamface.icon_state = "creampie_ai"
+
+		var/atom/A = parent
+		A.add_overlay(creamface)
+
+	pie_hit(source, stunning)

```

Comment:
This component has too many functions. The effects of the pie hitting shouldnt be tied to the graphics for it in component code.